### DISCUSSION TITLE
Should a Pi Zero 2W be this slow?

### DISCUSSION THREAD

**Discussion by EricWalklet** [retronas/discussions#50] [Q&A]
Hey, everyone.

I finally stopped procrastinating and started to install RetroNAS on my Raspberry Pi Zero 2W today. I'm running the 64-bit version of the OS and am using a USB-to-Ethernet adapter to connect it to my home network.

Everything seems to be taking a really long time. Installing a single service from the menu took 30-45 minutes. I then jumped ahead to install Cockpit (as Bob's video suggests) and that's been running for over an hour now with no sign of completing.

Is this normal? I'm new to the Pi scene, and I get that these things aren't Threadrippers, but I get the feeling something is wrong. Is this suspicious, or is setup supposed to be this sluggish if I'm using a low end Pi?

Thanks.

**Comment by sairuk** [comment]
That will be dependent on the services you are wanting to install, the Zero is going to be underpowered compared to other models but can theoretically run basic stuff.

some installers have grown in complexity since Bobs video as well, so they take longer now.

We do however recommend using a pi4 (even a pi3b+ is stretching it)

**Comment by EricWalklet** [comment]
I ended up speeding things up again by dropping the desktop environment from the Pi installer and doing everything from SSH, but it was still taking upwards of 30-60 minutes for each installer in the best case and completely crashing out in the worst.

I found an old Mac Mini in my box of random tech toys and have switched to that instead with much better results.

**Reply by danmons** [reply]
The PiZero2W is roughly the same generation CPU as the Pi3B with an ARM Cortex A53 CPU, although it's a slightly lower SKU and a much slower clockspeed, so performance is considerably lower (better than an RPi2B, worse than an RPi3B). 

A lot of RetroNAS install time performance is also about how fast you can write to the disk for the installs (lots of packages being decompressed and written to disk, lots of compiling of things depending on the tools you select).  If you're coupling the device with a very slow MicroSD card, that's going to slow down installs quite a lot. 

For the size and power draw, it's probably a nice little device.  But I don't think it's really got the power to serve much outside of VERY old equipment.
